### PR TITLE
feat(hot-toast): add support for context and injector

### PR DESCRIPTION
**Pull request by shhdharmen**
#49

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #49

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/ngneat/hot-toast/src/lib/components/hot-toast/hot-toast.component.ts [hunk 1]
```diff
@@ -93,6 +93,10 @@ export class HotToastComponent implements AfterViewInit, OnDestroy {
     return typeof this.toast.icon === 'string';
   }
 
+  get context() {
+    return this.toast.context ? { ...this.toast.context, toast: this.toastRef } : { $implicit: this.toastRef };
```
Comment (NetanelBasal) [inline projects/ngneat/hot-toast/src/lib/components/hot-toast/hot-toast.component.ts:97]: It is better to do it once on init

File: projects/ngneat/hot-toast/src/lib/hot-toast.model.ts [hunk 2]
```diff
@@ -150,6 +151,24 @@ export interface Toast {
...
   persist?: ToastPersistConfig;
+
+  /**
+   * Allows you to pass context for your template
+   *
+   * @since 1.1.0
+   * @type {*}
+   * @memberof Toast
+   */
+  context?: any;
```
Comment (NetanelBasal) [inline projects/ngneat/hot-toast/src/lib/hot-toast.model.ts:162]: Should be Record

File: projects/ngneat/hot-toast/src/lib/components/hot-toast/hot-toast.component.ts (hunk 1 above)
Comment (NetanelBasal) [inline projects/ngneat/hot-toast/src/lib/components/hot-toast/hot-toast.component.ts:97]: Why change the implicit when you have context?

Comment (shhdharmen) [reply projects/ngneat/hot-toast/src/lib/components/hot-toast/hot-toast.component.ts:97]: It's because user can pass their own `$implicit`.

Comment (NetanelBasal) [reply projects/ngneat/hot-toast/src/lib/components/hot-toast/hot-toast.component.ts:97]: IMO we should be consistent and always expose the toast under $implicit. We can mention in the docs that it's reserved for this.

**Review by NetanelBasal — APPROVED** [review]

**Pull request merged**